### PR TITLE
Split filter items to avoid BSON document too large if source repo contains > 345,000 units of same type

### PR DESCRIPTION
**Pull request by zxiong**
The issue was caused by long query (>16M) if source repo contains too many
units ( > 345,000) of same type.

The PR splits unit_ids in the filter to several chunks, then fetches data
separately, eventually combines all results.

Note that the PR only handle the case when the query does not need to sort.
It means the PR will not help if the query contains sorting. So the patch does
not fix the issue pulp#2220 completely.

ref pulp#2220
https://pulp.plan.io/issues/2220

### CODE REVIEW

**Comment by rbikar** [comment]
@zxiong  Looks good to me, I'd rather leave +1 for pulp developer as I'm not that familiar with this part of code.

**Comment by dkliban** [comment]
ok test

**Comment by daviddavis** [comment]
This patch looks good to me. I've asked the Pulp team for a second review since I haven't worked much in this area of Pulp 2.

Also, I think we'll probably wait until @ipanova is back in January to merge this since we don't have any upcoming releases scheduled before her return.

File: server/pulp/server/managers/repo/unit_association_query.py
Comment (ipanova) [inline server/pulp/server/managers/repo/unit_association_query.py]: i would rather name this variable units_batch_size, chunk can be confusing

File: server/pulp/server/managers/repo/unit_association_query.py
Comment (ipanova) [inline server/pulp/server/managers/repo/unit_association_query.py:129]: if you could leave here a comment with an explanation something along the lines https://pulp.plan.io/issues/2220#note-2 that would be very helpful whenever we'd look into this code in the future.

**Comment by ipanova** [comment]
@zxiong thank you for the patch, it looks good, i left 2 minor comments.

File: server/pulp/server/managers/repo/unit_association_query.py
Comment (zxiong) [reply server/pulp/server/managers/repo/unit_association_query.py]: Replaced UNITS_CHUNK_SIZE with UNITS_BATCH_SIZE.

File: server/pulp/server/managers/repo/unit_association_query.py
Comment (zxiong) [reply server/pulp/server/managers/repo/unit_association_query.py:129]: Added comments to explain why and how to change.

**Comment by ipanova** [comment]
ok test

**Review by ipanova — APPROVED** [review]

**Pull request merged**
